sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpMessageConverter.cs

```diff
@@ -155,7 +155,7 @@ private static ArraySegment<byte> ReadStreamToArraySegment(Stream stream)
 
         public static AmqpMessage SBMessageToAmqpMessage(SBMessage sbMessage)
         {
-            var body = new ArraySegment<byte>((sbMessage.Body.IsEmpty) ? Array.Empty<byte>() : sbMessage.Body.ToArray());
+            var body = new ArraySegment<byte>((sbMessage.Body.AsBytes().IsEmpty) ? Array.Empty<byte>() : sbMessage.Body.AsBytes().ToArray());
```

Comment:
Does it make sense to hoist the `AsBytes()` call here so that we don't have to call it multiple times?   I can't remember if it performs any work/casting/allocations or is just a pass through to a member.

It may be worth considering using a `static readonly` empty array reference here as well, because `Array.Empty` allocates each call.  The AMQP library is confirmed to not mutate the body array, so a shared empty should be fine.

Reply:
As far as the shared empty array, we are allocating whenever there actually is data, which would be the majority of cases, so the benefit would be pretty limited. I wonder if it is worth the risk of exposing us to some downstream issue that might occur in AMQP library where for some reason the array gets mutated after the library thinks everyone is done with the body. Also, I noticed EH allocates for an empty array (just to add in the benefit of consistency to leaving it as is).